### PR TITLE
Fix EBUSY errors in loopback device test cleanup

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to #2427

### CODE REVIEW

**Comment by jbaublitz** [comment]
@mulkieran Okay excellent! This is ready for review. Proof that the retries are working is [here](https://github.com/stratis-storage/stratisd/runs/2058419611?check_suite_focus=true#step:9:375).

**Review by mulkieran — CHANGES_REQUESTED** [review]
Just one change...then this should be ready to go.

File: src/engine/strat_engine/tests/util.rs [hunk 1]
```diff
@@ -45,13 +45,39 @@ fn dm_stratis_devices_remove() -> Result<()> {
...
                 }
             }
         }
+
+        // Retries if no progress has been made.
+        if !remain.is_empty() && !progress_made {
+            remain = remain
+                .into_iter()
+                .filter(|name| {
+                    let dm_name = match DmName::new(&name) {
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/tests/util.rs]: Resolved.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Looks good! Please close and open a final PR.

**Pull request closed without merging**